sources/packages/backend/apps/api/src/services/institution/institution.service.ts

```diff
@@ -783,21 +783,22 @@ export class InstitutionService extends RecordDataModelService<Institution> {
   async updateInstitution(
     institutionId: number,
     auditUserId: number,
-    updateInstitution: Partial<UpdateInstitution>,
+    updateInstitution: UpdateInstitution,
+    options?: { allowFullUpdate: boolean },
   ): Promise<Institution> {
     const institution = new Institution();
     institution.id = institutionId;
-
-    institution.operatingName = updateInstitution.operatingName;
-    institution.primaryPhone = updateInstitution.primaryPhone;
-    institution.primaryEmail = updateInstitution.primaryEmail;
-    institution.website = updateInstitution.website;
-    institution.regulatingBody = updateInstitution.regulatingBody;
-    institution.establishedDate = updateInstitution.establishedDate;
-    institution.institutionType = {
-      id: updateInstitution.institutionType,
-    } as InstitutionType;
-
+    if (options?.allowFullUpdate) {
```

Comment:
mm...what was the intention here

Reply:
Update institutions can be of 2 types.
- Institution updating institution profile where only Institution contact details and mailing address only can be updated.
- Ministry updating Institution profile where everything(except legal operating name) can be updated and considered as full update.